core/eth/reader.go

```diff
@@ -37,6 +38,7 @@ type ContractBindings struct {
 	EigenDAServiceManager *eigendasrvmg.ContractEigenDAServiceManager
 	EjectionManager       *ejectionmg.ContractEjectionManager
 	AVSDirectory          *avsdir.ContractAVSDirectory
+	PaymentVault          *paymentvault.ContractPaymentVault
```

Comment:
should this be initialized in `updateContractBindings`?

Reply:
Cool, are you planning to leave this PR open till the contract code is ready and deployed?
Maybe we should just add nil checks and proceed for now? We can update/remove the nil checks when contracts are ready

Reply:
I added nil checks before the contract calls so we can go ahead to merge! I think the payment DASM link is almost ready, can work on that as soon as that one gets merged